client/components/singleentry/DetailTable.js

```diff
@@ -2,10 +2,63 @@ import React from 'react'
 
 export default class DetailTable extends React.Component {
 
+  constructor (props) {
+    super(props)
+
+    var entries = []
+
+    for (var item in props.myData) {
+      if (item !== 'tags') {
+        entries.push({
+          name: item,
+          value: props.myData[item]
+        })
+      }
+    }
+
+    if (props.myData['tags']) {
+      var i = 0
+      entries.push({
+        name: 'Tags',
+        value: props.myData.tags.map(name => {
+          var clickHandler = function () { this.context.router.transitionTo('/search/' + name) }.bind(this)
+          return (
+            <span key={i++} onClick={clickHandler}><div className='label label-default'>{name}</div> </span>
+          )
+        })
+      })
+    }
+
+    this.state = {
+      entries: entries
+    }
+  }
+
   render () {
+    var i = 0

```

Comment:
Same here. i as the key in map function is limited inside `render`, so probably `let` is better